src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,182 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var extendFlat = require('../../lib/extend').extendFlat;
+var colorAttributes = require('../../components/colorscale/attributes');
+
+var scatterAttrs = require('../scatter/attributes');
+var scatterMarkerAttrs = scatterAttrs.marker;
+var colorbarAttrs = require('../../components/colorbar/attributes');
+
+var marker = extendFlat({
+    editType: 'calc'
+}, colorAttributes('marker', {editType: 'calc'}),
+    {
+        showscale: scatterMarkerAttrs.showscale,
+        colorbar: colorbarAttrs,
+        shape: {
+            valType: 'enumerated',
+            values: ['straight', 'curved'],
+            dflt: 'curved',
+            role: 'info',
+            editType: 'plot',
+            description: 'Sets the shape of the paths'},
+    });
+
+module.exports = {
+    domain: {
+        x: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1, editType: 'calc'},
+                {valType: 'number', min: 0, max: 1, editType: 'calc'}
+            ],
+            dflt: [0, 1],
+            editType: 'calc',
+            description: [
+                'Sets the horizontal domain of this `parcats` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        y: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1, editType: 'calc'},
+                {valType: 'number', min: 0, max: 1, editType: 'calc'}
+            ],
+            dflt: [0, 1],
+            editType: 'calc',
+            description: [
+                'Sets the vertical domain of this `parcats` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        editType: 'calc'
+    },
+
+    tooltip: {
```

Comment:
this should be handled by [`hoverinfo`](https://github.com/plotly/plotly.js/blob/0c1d6d5ebbd2334f712aede2a32236fb451838b7/src/traces/sankey/attributes.js#L21)

Reply:
Sounds good. I'll work through the comparison with Sankey, but I'm thinking the flaglist would be a combination of 'count' and/or 'probability', or 'none' to disable it all together.

Reply:
Done in 3680084

`tooltip` is gone and `hoverinfo` is added. As usual, `hoverinfo` can be one of `skip`, `none`, or `all`. Or a combination of `count` and `probability`.

Having the `skip` `hoverinfo` value also removed the need for the `none` `hovermode` enumeration value.